### PR TITLE
[E-node < T498] Implement relationship_exists

### PR DESCRIPTION
**Pull request by ind1xa**
### Description

Implemented node.relationship_exists procedure.

### Pull request type

- [ ] Bugfix
- [x] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/node_module/algorithm/node.cpp [hunk 1]
```diff
@@ -1 +1,43 @@
...
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    const auto node = arguments[0].ValueNode();
+    const auto pattern = arguments[1].ValueString();
+    bool exists = false;
+
+    if (pattern[0] == '<' && pattern[pattern.size() - 1] == '>') {
+      throw mgp::ValueException("Invalid relationship specification!");
+    }
```
Comment (antepusic) [inline cpp/node_module/algorithm/node.cpp:26]: This should be consistent with `node.relationship_types` (#300) and `nodes.relationship_types` (#301) - I think the <> case is the same as the bidirectional relationship case.

File: cpp/node_module/node_module.cpp [hunk 2]
```diff
@@ -5,6 +5,10 @@
 extern "C" int mgp_init_module(struct mgp_module *module, struct mgp_memory *memory) {
   try {
     mgp::memory = memory;
+    AddProcedure(Node::RelExists, Node::kProcedureRelExists, mgp::ProcedureType::Read,
+                 {mgp::Parameter(Node::kArgumentsNode, mgp::Type::Node),
+                  mgp::Parameter(Node::kArgumentsPattern, mgp::Type::String, "")},
```
Comment (antepusic) [inline cpp/node_module/node_module.cpp:10]: We should also accommodate the case where multiple patterns are given (#300 and #301 use a list for the second parameter).

**Review by antepusic — CHANGES_REQUESTED** [review]
The code looks good! However, there are two cases where this procedure should behave consistently with #300 and #301 and I've left comments on that.

File: cpp/node_module/algorithm/node.cpp (hunk 1 above)
Comment (ind1xa) [reply cpp/node_module/algorithm/node.cpp:26]: We decided to throw error in this case

File: cpp/node_module/node_module.cpp (hunk 2 above)
Comment (ind1xa) [reply cpp/node_module/node_module.cpp:10]: That is the only difference between node.relationship_exists and node.relationships_exist, if I do this, those will be the same procedures right? https://github.com/memgraph/mage/pull/298/files

File: cpp/node_module/algorithm/node.cpp (hunk 1 above)
Comment (antepusic) [reply cpp/node_module/algorithm/node.cpp:26]: I see, thanks!

File: cpp/node_module/node_module.cpp (hunk 2 above)
Comment (antepusic) [reply cpp/node_module/node_module.cpp:10]: The return would still be different. `relationships_exist` returns the existence of a relationship for _each_ of the given patterns, whereas this method returns whether there is a relationship that fits `at least one` of the given pattern(s).

**Review by antepusic — APPROVED** [review]

**Pull request merged**